torch/utils/data/datapipes/iter/selecting.py

```diff
@@ -63,6 +65,7 @@ def __init__(
         self.drop_empty_batches = drop_empty_batches
 
         self.input_col = input_col
+        # ensure_map_fn_works(filter_fn, self.input_col)
```

Comment:
Can I know which test is failing?

Reply:
test_datapipe.py (the logs did not specify which test failed)

Reply:
Looks like it also broke backwards-compat somehow, let me rebase.